### ISSUE TITLE
KissXML requires Swift without needing it

### ISSUE THREAD

**Issue by akempgen** [robbiehanson/KissXML#92]
Currently KissXML requires Swift, even when the containing app as well as KissXML itself are written only in Objective C. `otool -L KissXML` reveals this: 

    @rpath/libswiftCoreGraphics.dylib (compatibility version 1.0.0, current version 802.0.53)
    @rpath/libswiftDarwin.dylib (compatibility version 1.0.0, current version 802.0.53)
    @rpath/libswiftDispatch.dylib (compatibility version 1.0.0, current version 802.0.53)
    @rpath/libswiftFoundation.dylib (compatibility version 1.0.0, current version 802.0.53)
    @rpath/libswiftIOKit.dylib (compatibility version 1.0.0, current version 802.0.53)
    @rpath/libswiftObjectiveC.dylib (compatibility version 1.0.0, current version 802.0.53)

The only reason for requiring swift seems to be the DDXML.swift file, which was apparently added for nicer imports to Swift. But nicer Swift names can also be accomplished without forcing a Swift dependency on Objective C projects: https://developer.apple.com/library/content/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html#//apple_ref/doc/uid/TP40014216-CH10-ID162

**Comment by chrisballinger** [comment]
Unfortunately that breaks backwards compatibility with existing Swift projects, requiring everyone to rename everything.

**Comment by chrisballinger** [comment]
I took another stab at this but it was a lot harder than I anticipated. If someone can figure out a nice, clean way to do it, I'll gladly merge.

**Comment by fullsailor** [comment]
I'm now doing this in my Podfile as I was also getting a build error saying that the project wasn't specifying a SWIFT_VERSION.

```
pre_install do | installer |
  require 'fileutils'
  # Remove KissXML Swift support
  FileUtils.rm_rf(installer.sandbox.pod_dir('KissXML') + "KissXML/DDXML.swift")
end
```
It's not very robust. If KissXML adds more swift sources, I'll probably update it to use a "*.swift" glob.

**Comment by mstarke** [comment]
I took another try at this but ran into a naming Problem with getting the typealias carried over to Swift to be able to differentiate between NSXML and DDXML.

Adding a typedef ```typedef DDXMLDocument XMLDocument;``` does not get carried over to Swift.
Using ```@compatibility_alias XMLDocument DDXMLDocument;``` creates a rather unwanted typealias ```public typealias XMLDocument = XMLDocument```

Adding NS_SWIFT_NAME to all the interfaces resolves all other issues but this one is out of my comfort zone since I'm laking basic Swift skills.

**Comment by mstarke** [comment]
Removing ```NS_SWIFT_NAME``` on the classes and only using ```@compatibilty_alias``` seems to do the trick. I created PR #103 with the solution.

**Comment by chrisballinger** [comment]
Swift is no longer required (resolved by #103)
